src/WebJobs.Script/Description/Rpc/WorkerFunctionDescriptorProvider.cs

```diff
@@ -40,5 +43,27 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
             });
             return new WorkerLanguageInvoker(Host, triggerMetadata, functionMetadata, inputBindings, outputBindings, inputBuffer);
         }
+
+        protected override Collection<ParameterDescriptor> GetFunctionParameters(IFunctionInvoker functionInvoker, FunctionMetadata functionMetadata,
+            BindingMetadata triggerMetadata, Collection<CustomAttributeBuilder> methodAttributes, Collection<FunctionBinding> inputBindings, Collection<FunctionBinding> outputBindings)
+        {
+            var parameters = base.GetFunctionParameters(functionInvoker, functionMetadata, triggerMetadata, methodAttributes, inputBindings, outputBindings);
+
+            var bindings = inputBindings.Union(outputBindings);
+
+            var returnsSomething = bindings.SingleOrDefault(b => b.Metadata.ReturnsSomething);
+            if (returnsSomething != null)
+            {
+                Type type = typeof(Task<object>);
+                var type2 = type.GetGenericArguments()[0];
+                var byRefType = type2.MakeByRefType();
```

Comment:
Unless your something is missing here and this wasn't your intention, the above lines can be simplified and just replaced with `typeof(object).MakeByRefType()`

Reply:
Should this be JObject?   

---
In reply to: [146417350](https://github.com/Azure/azure-webjobs-sdk-script/pull/2049#discussion_r146417350) [](ancestors = 146417350)